radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/Forks.java

```diff
@@ -147,9 +144,9 @@ public static Forks create(Set<ForkConfig> forks) {
     return new Forks(fixedEpochForks, maybeCandidateFork);
   }
 
-  private static boolean ensureUniqueHashes(Set<ForkConfig> forks) {
-    final var hashesSet = forks.stream().map(ForkConfig::hash).collect(Collectors.toSet());
-    return forks.size() == hashesSet.size();
+  private static boolean ensureUniqueNames(Set<ForkConfig> forks) {
+    final var namesSet = forks.stream().map(ForkConfig::name).collect(Collectors.toSet());
```

Comment:
Still in the topic of the Hungarian Notation, I believe it is mostly considered a bad practice https://en.wikipedia.org/wiki/Hungarian_notation#Notable_opinions.

Reply:
I think it's justified in this particular case, using Set specifically is the mechanism that actually checks for duplicates. Calling it simply `names` would provide much less context imo

Reply:
But what the real value of doing that in a statically typed language? If a developer wants to know the type of a variable it can be easily found.

Reply:
the real value is that it explicitly states that the variable is a set, and it must remain a set because this is a part of the implementation, if the name was just `names` it wouldn't be obvious that it can't just be replaced with a list

Reply:
I've changed it to a more descriptive `namesWithoutDuplicates` that isn't tied to a specific type